services/sources/publisher_backlist_service.py

```diff
@@ -0,0 +1,115 @@
+        limit: Optional[int]=None
+    ) -> list[UofMMapping]:
+        
+        array_json_records = self.get_records_json(full_import, start_timestamp, offset, limit)
+
+        array_index = 0
+        while array_index != len(array_json_records):
+            for dict_index in range(0, len(array_json_records[array_index]['records'])):
+                try:
+                    record_metadata_dict = array_json_records[array_index]['records'][dict_index]
+                    record = UofMMapping(record_metadata_dict)
+                    record.applyMapping()
+                except Exception as e:
+                    logger.exception(f'Failed to process Publisher Backlist record due to {e}')
+
+            array_index += 1
+        
+
+
+    def get_records_json(self,
+        full_import: bool=False, 
+        start_timestamp: datetime=None,
+        offset: Optional[int]=None,
+        limit: Optional[int]=None
+    ) -> list[dict]:
+
+        if offset == None:
+            limit = 100
+
+        headers = {"Authorization": f"Bearer {self.airtable_auth_token}"}
+
+        if not full_import:
+            if start_timestamp:
+
+                filter_by_formula = self.build_filter_by_formula_parameter(start_timestamp)
+
+                array_json_records = self.get_records_array(headers, limit, filter_by_formula)
+
+                return array_json_records
+
+            else:
+
+                filter_by_formula = self.build_filter_by_formula_parameter(start_timestamp)
+
+                array_json_records = self.get_records_array(headers, limit, filter_by_formula)
+
+                return array_json_records
+            
+        else:
+
+            array_json_records = self.get_records_array(headers, limit, filter_by_formula=None)
+        
+            return array_json_records
+        
+    def build_filter_by_formula_parameter(self, start_timestamp: datetime=None) -> str:
+        if not start_timestamp:
+            start_timestamp = datetime.now(timezone.utc).replace(tzinfo=None) - timedelta(hours=24)
+
+        start_date_time_str = start_timestamp.strftime("%Y-%m-%d %H:%M:%S.%f")
+        start_date_time_encoded = urllib.parse.quote(start_date_time_str)
+        is_same_date_time_filter = f"IS_SAME(%7BLast%20Modified%7D,%20%22{start_date_time_encoded}%22,%20%22second%22"
+        is_after_date_time_filter = f"%20IS_AFTER(%7BLast%20Modified%7D,%20%22{start_date_time_encoded}%22"
+        
+        filter_by_formula = f"OR({is_same_date_time_filter}),{is_after_date_time_filter}))"
+        return filter_by_formula
+        
+    def get_records_array(self, 
+        headers: Dict[str, str], 
+        limit: Optional[int]=None, 
+        filter_by_formula: str=None,
+    ) -> list[dict]:
+        
+        if filter_by_formula: 
+            url_format = f'{BASE_URL}&filterByFormula={filter_by_formula}&pageSize={limit}'
+        else: 
+            url_format = f'{BASE_URL}&pageSize={limit}'
```

Comment:
looks like you can simplify this because we are adding the pageSize in both cases. 

```
url = f'{BASE_URL}&pageSize={limit}

if filter_by_formula:
  url += f'&filterByFormula{filter_by_formula}'